### PR TITLE
Make decor mandatory and labels optional.

### PR DESCRIPTION
**Pull request by thorstenhater**
- Make decor mandatory 
- ...and label arguments to cable cell optional
- Put the most optional argument (labels) last.

### CODE REVIEW

**Comment by brenthuisman** [comment]
Tests are perhaps not what defaults should be based on. And I can't imagine a simulation of cells without any mechanisms. So unless there is an actual use case, I would like to require it such that people don't forget that cells without e.g. mechanisms are almost certainly not what they want.

**Comment by thorstenhater** [comment]
They could have always done that so far, but sure, I see the point. More breaking change then ;)

**Comment by thorstenhater** [comment]
Done.

**Review by boeschf — COMMENTED** [review]
Looks good - these small changes have quite a substantial ripple effect!
Please merge with master, there are a couple of recipes in the SDE tests that need to be updated.

**Comment by brenthuisman** [comment]
@thorstenhater If you could merge master and address @boeschf 's comments, we can merge these breaking changes before v0.8.

File: python/cells.cpp [hunk 1]
```diff
@@ -914,17 +914,27 @@ void register_cells(pybind11::module& m) {
         "tree of one-dimensional cable segments.");
     cable_cell
         .def(pybind11::init(
-            [](const arb::morphology& m, const label_dict_proxy& labels, const arb::decor& d) {
-                return arb::cable_cell(m, labels.dict, d);
+            [](const arb::morphology& m, const std::optional<arb::decor>& d, const std::optional<label_dict_proxy>& l) {
```
[the remark this answers is not in the document]
Comment (thorstenhater) [reply python/cells.cpp:917]: Maybe yes, polishing the UX is always appreciated, but I wouldn't stuff that into this PR.

File: doc/python/cable_cell.rst [hunk 2]
```diff
@@ -52,14 +52,14 @@ Cable cells
...
         # Construct a cable cell.
-        cell = arbor.cable_cell(morph, labels, decor)
+        cell = arbor.cable_cell(morph, decor, labels)
 
-    .. method:: __init__(morphology, labels, decorations)
+    .. method:: __init__(morph, decorations, labels)
 
         Constructor.
 
         :param morphology: the morphology of the cell
```
[the remark this answers is not in the document]
Comment (thorstenhater) [reply doc/python/cable_cell.rst:61]: True.

**Comment by thorstenhater** [comment]
Done, I fixed two new warnings in `test_sde` along with it.
- `sde_recipe& add_probe(probe_tag tag, std::any address)` hides a virtual on its base class
- `auto sampler_ = [ncells, nsteps] (std::vector<arb_value_type>& results, unsigned count,` doesn't use its `ncells` capture.

**Comment by boeschf** [comment]
there is one more recipe to fix in `test_sde`, line 826 (only enabled when compiling for GPU)

**Comment by thorstenhater** [comment]
Also, there's the `malloc_consolidate: invalid chunk size` again.

**Comment by thorstenhater** [comment]
bors try

**Comment by brenthuisman** [comment]
bors try

**Comment by brenthuisman** [comment]
`ci/release/build.Dockerfile` contains a hardcoded URL to CMake 3.16 (Bors failed on that line, not the tests this time). Since elsewhere there was a bump, we could do so here too.

**Comment by thorstenhater** [comment]
That file hasn't changed in a while and I have this:
```
RUN wget -q "https://github.com/Kitware/CMake/releases/download/v3.18.6/cmake-3.18.6-Linux-x86_64.tar.gz" -O cmake.tar.gz && \
    echo "87136646867ed65e935d6bacd44d52a740c448ad0806f6897d8c3d47ce438c8b cmake.tar.gz" | sha256sum --check --quiet && \
    tar --strip-components=1 -xzf cmake.tar.gz -C /usr/local && \
    rm -rf cmake.tar.gz
```

**Review by boeschf — APPROVED** [review]

**Pull request merged**